pallets/runtime/src/general_tm.rs

```diff
@@ -136,58 +155,76 @@ decl_module! {
         fn deposit_event() = default;
 
         /// Adds an asset rule to active rules for a ticker
-        pub fn add_active_rule(origin, ticker: Ticker, asset_rule: AssetRule) -> DispatchResult {
+        ///
+        /// # Arguments
+        /// * origin - Signer of the dispatchable. It should be the owner of the ticker
+        /// * ticker - Symbol of the asset
+        /// * asset_rule - Sender & receiver rule.
+        pub fn add_active_rule(origin, ticker: Ticker, asset_rule: AssetRuleDetails) -> DispatchResult {
```

Comment:
Let's take `sender_rules` and `receiver_rules` as separate inputs rather than using `AssetRuleDetails`. That way, we won't have to clone them later to generate `AssetRule`.

Reply:
We still need to clone because it is a vector.

Reply:
But I think it is clear to use vec directly, I personally don't like the idea of extra struct to use as the param